### ISSUE TITLE
Google login fails

### ISSUE THREAD

**Issue by drewthor** [omarroth/invidious#154]
The method used for Google login causes google to block the login as label it as a critical security alert. I do have 2FA enabled.

**Comment by omarroth** [comment]
Unfortunately I'm having trouble replicating this. Currently Invidious only supports SMS or Authenticator apps, which may be the issue. Would it be possible to provide a screenshot with user info removed?

**Comment by aturn3** [comment]
<img width="678" alt="image" src="https://user-images.githubusercontent.com/9273455/45261138-2cb5ba00-b3c0-11e8-8db9-a7ac6be4f7f6.png">

Here is the security event. I left the IP out of the screenshot.

What exactly would you like a screenshot of?

**Comment by omarroth** [comment]
That was all I was interested in, thank you. Do you have either SMS or Google Authenticator as a form of authentication?

**Comment by aturn3** [comment]
Google authenticator

**Comment by omarroth** [comment]
Sorry for the delayed response. I'm unfortunately having trouble replicating this. Does it immediately fail? Or does it show a box for entering your authentication code? If it fails without asking you for a code, try logging in [here](https://invidio.us) and see if it works as expected.

If it still fails, I'd recommend making a new account and importing your subscriptions.

Apologies again fro the delay.

**Comment by peyloride** [comment]
I do have same problem. I have 2FA enabled, but can't login. Same as OP.

**Comment by omarroth** [comment]
Unfortunately it's very difficult for me to debug since I can't reproduce. You might try logging in manually on Google and making sure you don't encounter any extra verification or pages besides TFA. 

If you still can't login, would you mind contacting me at omarroth@hotmail.com or omarroth@protonmail.com? I would like to have either test it locally on my machine or help you set it up in order to find the issue.

**Comment by omarroth** [comment]
Added c7f0a6f2e1e8131bcfd3052caa185973fb18a4fb, which should fix the issue. Would you mind seeing if you can now login @aturn3?

**Comment by omarroth** [comment]
You might try logging in [here](https://invidio.us/login?type=google&tfa=true), to see if providing an authentication code allows you to login. Otherwise, would you mind contacting me privately?

**Comment by omarroth** [comment]
As an update to this, I've tested with several accounts with no TFA, SMS, and Authenticator and been able to login successfully, so I expect it's either a prompt or security method that Invidious can't handle or some other difference between accounts that breaks login.

For any issues with Google login I'm still happy to debug, although it is time-consuming. If possible testing with `youtube-dl -u $USERNAME -v https://www.youtube.com/feed/history` will provide some more info that should help find the issue. Feel free to contact me privately (@omarroth:matrix.org on Matrix or omarroth@protonmail.com) with more information.

Closing in favor of #292.
